src/utils/axiosSetup.ts

```diff
@@ -1,65 +1,91 @@
-import axios, { AxiosRequestConfig } from 'axios';
+import axios, { AxiosError, AxiosRequestConfig, AxiosResponse } from 'axios';
 
 import { AuthenticationEndpoint } from '../authentication';
+import { TokenResponse } from '../authentication/types';
 import { ApiState } from '../baseApi';
+import { ErrorFormat } from '../errors';
 import { ConfigProvider } from '.';
 
-type interceptorParams = {
+export type InterceptorParams = {
   state: ApiState;
   config: ConfigProvider;
   authentication: AuthenticationEndpoint;
 };
 
+const requestInterceptor = (config: AxiosRequestConfig) => {
+  // axios.defaults do not automatically apply here, so set timeout manually
+  config.timeout = axios.defaults.timeout;
+  return config;
+};
+
+const responseFulfilledInterceptor = (res: AxiosResponse) => res;
+
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export const interceptor = (apiInstanceData: interceptorParams) => {
-  axios.interceptors.response.use(
-    (res) => {
-      return res;
-    },
-    async (error) => {
-      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions,@typescript-eslint/no-unsafe-member-access
-      if (apiInstanceData.config.settings.autoRefreshToken && error.response) {
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment,@typescript-eslint/no-unsafe-member-access
-        const untouchedRequest = error.config;
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,@typescript-eslint/no-unsafe-call,@typescript-eslint/strict-boolean-expressions
-        if (error.response.status === 497 && untouchedRequest.url.includes('auth/oauth2/token')) {
-          if (apiInstanceData.config.settings.onTokenRefreshedFailed) {
-            apiInstanceData.config.settings.onTokenRefreshedFailed();
+export const responseRejectedInterceptor = (interceptorParams: InterceptorParams) => async (error: AxiosError) => {
+  // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions,@typescript-eslint/no-unsafe-member-access
+  if (interceptorParams.config.settings.autoRefreshToken && error.response) {
```

Comment:
Can we maybe rename this to `shouldAutoRefreshToken` or something similar? Doesn't look like a bool to me.